pkg/config/corscfg/flags.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2023 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package corscfg
+
+import (
+	"flag"
+
+	"github.com/spf13/viper"
+)
+
+const (
+	corsPrefix         = ".cors"
+	corsAllowedHeaders = corsPrefix + ".allowed-headers"
+	corsAllowedOrigins = corsPrefix + ".allowed-origins"
+)
+
+type Flags struct {
+	Prefix string
+}
+
+func (c Flags) AddFlags(flags *flag.FlagSet) {
+	flags.String(c.Prefix+corsAllowedHeaders, "content-type", "Allowed headers for the HTTP port , default content-type")
+	flags.String(c.Prefix+corsAllowedOrigins, "*", "Allowed origins for the HTTP port , default accepts all")
+}
+
+func (c Flags) InitFromViper(v *viper.Viper) Options {
+	var p Options
+	p.AllowedHeaders = v.GetStringSlice(c.Prefix + corsAllowedHeaders)
```

Comment:
I see that previously we were stripping whitespace
```
strings.Split(strings.ReplaceAll(params.AllowedOrigins, " ", ""), ",")
```
Is it going to be done automatically by `viper.GetStringSlice`? If not it's not a backwards compatible change.

Reply:
can you add tests to this package? it's small and has no external deps, no reason not to write tests, and in particular validate ^^^

Reply:
I just was checking with you if i was going directionally right ! Will write the tests and validate !

Reply:
I was just checking with you if I was going in the right direction , since I am new to the community. I will write the tests and validate soon, sorry for the delay!